adagrams/game.py

```diff
@@ -1,11 +1,104 @@
+from string import ascii_uppercase
+from random import randint , choice
+
+
+
 def draw_letters():
-    pass
+    letter_pool = {
+        'A': 9, 
+        'B': 2, 
+        'C': 2, 
+        'D': 4, 
+        'E': 12, 
+        'F': 2, 
+        'G': 3, 
+        'H': 2, 
+        'I': 9, 
+        'J': 1, 
+        'K': 1, 
+        'L': 4, 
+        'M': 2, 
+        'N': 6, 
+        'O': 8, 
+        'P': 2, 
+        'Q': 1, 
+        'R': 6, 
+        'S': 4, 
+        'T': 6, 
+        'U': 4, 
+        'V': 2, 
+        'W': 2, 
+        'X': 1, 
+        'Y': 2, 
+        'Z': 1
+    }
+    hand = []
+
+
+    while len(hand) < 10:
+        pick = choice(ascii_uppercase)
+        if letter_pool[pick] > 0:
+            hand.append(pick)
+            letter_pool[pick] -= 1
+    return hand
+
+    # while len(hand) < 10:
+    #     pick = ascii_uppercase[randint(0, len(ascii_uppercase)-1)]
+    #     if letter_pool[pick] > 0:
+    #         hand.append(pick)
+    #         letter_pool[pick] -= 1
+    # return hand
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    word_list = list(word.upper())
+    letter_bank_list = list(letter_bank)
+
+    for letter in word_list:
+        if letter in letter_bank_list:
+            letter_bank_list.remove(letter)
+        else:
+            return False
+    return True
+
 
 def score_word(word):
-    pass
+    SCORES_CHART = {
+    1: ['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T'],
+    2: ['D', 'G'],
+    3: ['B', 'C', 'M', 'P'],
+    4: ['F', 'H', 'V', 'W', 'Y'],
+    5: ['K'],
+    8: ['J', 'X'],
+    10: ['Q', 'Z']
+}
```

Comment:
Notice that when you have a nested data structure you also have nested for-loops. We cover Big O tomorrow so I didn't expect this to be a consideration when you wrote this code last week, but when your value in the dictionary is a list, you increase the time complexity of your solution. 

Looking up a value in a dictionary is constant time, but looking up a letter in a list is linear time. 

Using a dictionary like what's below would reduce the complexity here:
``` 
SCORES_CHART = {
  "A": 1, 
  "B": 3, 
  # etc...
}

Reply:
I agree. I was too lazy to retype the dictionary this way :sweat_smile:
I'll fix that

Reply:
Done